.cargo/config.toml

```diff
@@ -12,7 +12,74 @@ rustflags = [
 	# rustc additional warnings:
 	"--warn",
 	"unused_crate_dependencies",
-	# clippy additional warnings:
 	"--warn",
-	"clippy::wildcard_imports",
+	"rust_2018_idioms",
```

Comment:
I suggest replacing groups like `rust_2018_idioms` with its individual rules, since we are using both here. This makes it clear what is actually being used/enabled, and removes overlaps.
The mapping is here is up to date IIUC: https://doc.rust-lang.org/rustc/lints/groups.html

```
  "-Wbare-trait-objects",
  "-Welided-lifetimes-in-paths",
  "-Wellipsis-inclusive-range-patterns",
  "-Wexplicit-outlives-requirements",
  "-Wunused-extern-crates",
```

Reply:
Also, a few interesting groups I found in that page, if you think they are useful:

- `rust-2018-compatibility`
- `rust-2021-compatibility`
- `clippy::cargo`

Reply:
I'd prefer if we keep using the group in this case; warning against the lint group is *very* popular, see https://github.com/search?q=%23%21%5Bwarn%28rust_2018_idioms%29%5D&type=code.

About the `*-compatibility` lints, these are used when migrating *to* the given edition; given that we're already using Rust 2021, there's no value in enabling these. The 2018 one has heavy overlap with the 2018 idioms, which already are partially warn-by-default, with the rest being additionally warned here, wherease the 2021 warns against the changes to the `pat` matcher and different closure capture rules in 2021. Since we're using the 2021 edition for quite a bit, this is well-known and enforced already for some time in Slang.

Reply:
Now that I think about linting against groups, I think it makes sense to lint against specific Clippy lint groups but cherry-pick only from specific ones, i.e. restriction and pedantic. I'll try to do that now.

Reply:
I think that grouping makes sense if the groups don't overlap (like clippy categories). But for Rust, it doesn't, which makes it not obvious at all which are turned on vs which are not, and makes it even harder tracing through git history for changes.

Reply:
I think that familiarity and brevity here beats the slightly more correctness of only warning against the lints in the `rust_2018_idioms`, all of these lints are eventually planned to be warned by default, anyway, as part of the remaining edition push work.

However, this is minor and I think it's not worth spending extra time on discussing this; I'm happy to include the not-yet-warn-by-default from the group separately.

Reply:
Definitely a nit pick, not blocking in any way.